src/v/config/config_store.h

```diff
@@ -194,6 +205,11 @@ class config_store {
 private:
     friend class base_property;
     std::unordered_map<std::string_view, base_property*> _properties;
+
+    // If a property has some aliases for backward compat, they are tracked
+    // here: a property must appear at least in _properties, and may appear
+    // 0..n times in _aliases
+    std::unordered_map<std::string_view, base_property*> _aliases;
```

Comment:
i think it is true, but just to confirm: all of these string_views (here and back in meta) are all expected to be pointing at string literals?

Reply:
Right, they're all the string literals in the config::configuration/config::node_config constructors.  It's a shame we don't have a robust way to guarantee that, but it's the existing way that the _properties map works.